compose-table/src/androidTest/java/org/dhis2/composetable/TextInputUiTest.kt

```diff
@@ -0,0 +1,114 @@
+package org.dhis2.composetable
+
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.BottomSheetScaffold
+import androidx.compose.material.BottomSheetValue
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.rememberBottomSheetScaffoldState
+import androidx.compose.material.rememberBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.launch
+import org.dhis2.composetable.activity.TableTestActivity
+import org.dhis2.composetable.data.tableData
+import org.dhis2.composetable.model.TableCell
+import org.dhis2.composetable.model.TextInputModel
+import org.dhis2.composetable.ui.TableColors
+import org.dhis2.composetable.ui.TableList
+import org.dhis2.composetable.ui.TextInput
+import org.junit.Rule
+import org.junit.Test
+
+class TextInputUiTest {
+
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<TableTestActivity>()
+
+    @Test
+    fun myTest() {
+        var cellToSave: TableCell? = null
+        val expectedValue = "55"
+
+        composeTestRule.setContent {
+            TextInputUiTestScreen {
+                cellToSave = it
+            }
+        }
+
+        tableRobot(composeTestRule) {
+            assertClickOnCellShouldOpenInputComponent(0, 0)
+            assertClickOnEditOpensInputKeyboard()
+            assertClickOnSaveHidesKeyboardAndSaveValue(expectedValue)
+            assert(cellToSave?.value == expectedValue)
+        }
+    }
+
+    @OptIn(ExperimentalMaterialApi::class)
+    @Composable
+    private fun TextInputUiTestScreen(onSave:(TableCell)->Unit){
```

Comment:
can you give a more descriptive name? (conditions/asserts/should/ etc...)

Reply:
This is not a Test. This is the Test composable that is launched.